dev/src/document.ts

```diff
@@ -116,14 +118,13 @@ export class DocumentSnapshot {
    * if the document does not exist).
    */
   constructor(
-    ref: DocumentReference,
-    fieldsProto?: ApiMapValue,
+    ref: DocumentReference<T>,
+    readonly fieldsProto?: ApiMapValue,
```

Comment:
I am undecided whether this should be `fieldsProto` or `_ fieldsProto`. This is a public field for internal consumption.